dotcom-rendering/src/web/components/ShowHideContainer.importable.tsx

```diff
@@ -0,0 +1,99 @@
+import { css } from '@emotion/react';
+import { from, space, neutral, textSans } from '@guardian/source-foundations';
+import { ButtonLink } from '@guardian/source-react-components';
+import { useEffect, useState } from 'react';
+import { storage } from '@guardian/libs';
+
+type Props = {
+	sectionId: string;
+	overrideContainerToggleColour?: string;
+};
+
+type ContainerStates = { [id: string]: string };
+
+const showHideButtonCss = (
+	overrideContainerToggleColour: string | undefined,
+) => {
+	console.log(overrideContainerToggleColour);
+	return css`
+		color: ${neutral[46] || overrideContainerToggleColour};
+		${textSans.xsmall()};
+	`;
+};
+
+const showHideContainerCss = css`
+	margin-top: ${space[2]}px;
+	margin-right: 10px;
+	margin-bottom: ${space[2]}px;
+	position: relative;
+	align-items: bottom;
+
+	${from.wide} {
+		position: absolute;
+		top: 0;
+		right: 0;
+	}
+`;
+
+const getContainerStates = (): ContainerStates => {
+	const item = storage.local.get(`gu.prefs.container-states`);
+
+	if (!item) {
+		return {};
+	}
+
+	return item;
+};
+
+export const ShowHideContainer = ({
+	sectionId,
+	overrideContainerToggleColour,
+}: Props) => {
+	const [isClosed, setClosed] = useState(
+		getContainerStates()[sectionId] === 'closed',
```

Comment:
This looks great!

On quick question: do you know whether getting/setting localstorage is a blocking operation? If not, is there any risk of race conditions between here and ln.79?

Reply:
(or does `getContainerStates()` only run the first time the component is rendered?)

Reply:
Good question!

It is blocking, getContainerStates will be called once before the component has rendered, it does get called again on line 69 and modified on line 79 but I don't think there can be a race condition as those lines could only be accessed once the component has finished rendering

Reply:
that's really helpful, thanks!